### PR TITLE
feat: [DHIS2-9858] update user sync test

### PR DESCRIPTION
**Pull request by Sharmyn28**
This PR has:

- update functions to get the number of org units, programs, datasets, and program rules related to a specific user.
- update functions to get metadata and data download size
- fix user search field styles
- fix result table styles

_User search field_
image

_Test loading_
image

_Test result_
image

### CODE REVIEW

**Comment by mediremi** [comment]
I'll be away for two weeks so I'm requesting review from @dhis2/team-platform

File: src/pages/Synchronization/UserSyncTest/helper.js [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    const orgUnitPromises = []
+    const orgUnitSearch = []
+    orgUnits.map(orgUnit =>
+        orgUnitPromises.push(apiFetchOUSearch(dataEngine, orgUnit.id))
+    )
+
+    return Promise.all(orgUnitPromises).then(result => {
+        result.map(ouList => ouList.map(ou => orgUnitSearch.push(ou)))
+        return uniqBy(orgUnitSearch, 'id').length
+    })
```
Comment (amcgee) [inline src/pages/Synchronization/UserSyncTest/helper.js:95]: Better to do this sequentially:

```suggestion
    const ouLists = Promise.all(orgUnitPromises)
    ouLists.forEach(ouList => 
        ouList.forEach(ou => {
            orgUnitSearch.push(ou)
        })
    }
    return uniqBy(orgUnitSearch, 'id').length
```

**Review by amcgee — CHANGES_REQUESTED** [review]
Lots of good stuff in here @Sharmyn28 

I found a number of patterns which make the code very difficult to read - I tried to make suggestions for most of them, but there are more which also follow these patterns.  In general, try to remember that:
1. `await` will return the value of the resolved promise, so you can assign that value to a normal variable in the async function and operate on it there, instead of using `then` and assigning a `let` variable
2. It is better to use `const` wherever possible, to avoid accidentally mutating something you don't want to change.
3. `Array.map` will return a new copy of the array with the map function applied to each item.  This is useful because you can then assign the result of `.map` to a new variable and just use that, instead of manually `.push`ing to another array.
4. When *not* assigning the result of a `map` to some other variable, it is better to use `forEach`.  This should generally be avoided as it creates deeply-nested code, but is more correct than using `map` and throwing away the return value.

Please let me know if you have any questions of want to go over any of this with me or someone else on the team!

File: src/pages/Synchronization/UserSyncTest/getData.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    orgUnit.map(ou =>
+        programList.map(program => {
+            const tei = program.teiDownload
+            let pageSize
+            if (tei) {
+                pageSize = limitByProgram
+                    ? globalLimit.teiDownload
+                    : Math.round(globalLimit.teiDownload / programList.length)
+            } else {
+                pageSize = limitByProgram
```
[the remark this answers is not in the document]
Comment (Sharmyn28) [reply src/pages/Synchronization/UserSyncTest/getData.js:36]: I'm trying to do this: 
```

if (limitByProgram) {
  pageSize = globalLimit.teiDownload
} else {
  pageSize = Math.round(globalLimit.teiDownload / programList.length)
}
```

but using a ternary operator

[the remark this answers is not in the document]
Comment (amcgee) [reply src/pages/Synchronization/UserSyncTest/getData.js:37]: Ah I see!  Maybe we could wrap the ternary in parenthesis to make it clearer?  But if prettier automatically removes those that's ok, this is fine as-is - sorry I misread it!

File: src/pages/Synchronization/UserSyncTest/getData.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+const getTEI = ({ dataEngine, programList, globalLimit, orgUnit }) => {
+    const teiPromises = []
+    const limitByProgram = globalLimit.settingDownload === 'PER_PROGRAM'
+    orgUnit.map(ou =>
+        programList.map(program => {
+            const tei = program.teiDownload
+            let pageSize
+            if (tei) {
+                pageSize = limitByProgram
```
[the remark this answers is not in the document]
Comment (amcgee) [reply src/pages/Synchronization/UserSyncTest/getData.js:33]: Same comment as above, sorry I misread this one!

File: src/pages/Synchronization/UserSyncTest/queries/syncQueries.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+})
+
+export const apiFetchOU = async (dataEngine, ou) => {
+    try {
+        const ouData = await dataEngine.query({
+            ou: orgUnitQuery(ou),
+        })
+        return ouData.ou.organisationUnits
+    } catch (error) {
+        console.log('Error: ', error)
```
[the remark this answers is not in the document]
Comment (amcgee) [reply src/pages/Synchronization/UserSyncTest/queries/syncQueries.js:41]: This is an important one, but I don't think it is new in this PR.  Maybe you can look at this in the future (let me know if you want to discuss) but keep it as-is for now :-)

**Review by amcgee — APPROVED** [review]
Really nice work, the updated `await` lines are much easier to understand!  Thanks!

I'm approving this now - if you want to wrap the ternary operators in parentheses to make that logic clearer, feel free.  And I think we should address the error handling (`console.log` in catch handlers) but I don't think it's new in this PR - if that is true, then it's OK to merge this and address that issue separately 👍 💯

**Pull request merged**